corehq/apps/translations/views.py

```diff
@@ -101,11 +101,11 @@ def _generate_translations_for_po(self, worksheet):
     def _generate_po_file(self, worksheet):
         """
         extract translations from worksheet and converts to a po file
-        :return: PoFileGenerator object
+        :return: list of files generated
         """
         translations = self._generate_translations_for_po(worksheet)
-        po_file_generator = PoFileGenerator(translations, {})
-        return po_file_generator
+        with PoFileGenerator(translations, {}) as po_file_generator:
+            return po_file_generator.generated_files[0][1]
```

Comment:
The generated file will be cleaned up before it is returned. To get around that you could read the content in this method before cleanup happens:
```py
def _generate_po_content(self, worksheet):
    """Extract translations from worksheet and convert to PO file content

    :return: UTF-8 string of PO file content
    """
    translations = self._generate_translations_for_po(worksheet)
    with PoFileGenerator(translations, {}) as po_file_generator:
        file_path = po_file_generator.generated_files[0][1]
        with open(file_path, 'r', encoding="utf-8") as f:
            return f.read()
```
the calling code would look like this
```py
content = self._generate_po_content(worksheet)
```

Reply:
Aha! thanks.. missed this.